lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -451,31 +451,68 @@ def change_indexes(model, old_table_name, to_rename)
               end
           end
 
-          drop_indexes = (existing_indexes_without_primary_key - model_indexes_without_primary_key).map do |i|
+          indexes_to_drop = existing_indexes_without_primary_key - model_indexes_without_primary_key
+          indexes_to_add = model_indexes_without_primary_key - existing_indexes_without_primary_key
+
+          renamed_indexes_to_drop = []
+          renamed_indexes_to_add = []
+
+          indexes_to_drop.each { |index_to_drop|
+            renamed_columns = index_to_drop.columns.map do |column|
+              to_rename[column]
+            end
+
+            indexes_to_add.each { |index_to_add|
+              if renamed_columns == index_to_add.columns
```

Comment:
These are both arrays, so comparing for equality will also check that the column order matches. Is there a chance that the set of columns matches, but the order doesn't (so `==` fails)? If there's a chance of that, you could always put them each into a `Set` first and then compare the sets.

Reply:
Good point. I'll look into it.

Reply:
I ended up going with the `Set` comparison. I'm not 100% certain that we need it, but the code is quite complex and I feel I could easily be overlooking something. I'm not aware of any reason using `Set` here would be a bad idea — and to me it seems like it would make the code a bit more resilient.